install.bat

```diff
@@ -1,25 +1,164 @@
-
-:: Windows: Creates a yolo.bat file into %HOME% directory which will let you run similar to Linux/MacOS
-:: Example: Input: ".\yolo.bat print hello" to "yolo print hello"
-
-:: yolo.bat can only be used in same directory; or, any directory if put in a $PATH directory (type $env:PATH in PowerShell and pick an appropriate path to paste in)
-:: C:\Windows\System32 is the $PATH directory everyone is likely to have
-
-:: Create yolo.bat and if it isn't already there input its code.
-find "@echo off" "%HOME%\yolo.bat" && (
-    echo "yolo.bat" Already Exists
-) || (
-    copy /y nul %HOME% yolo.bat
-    echo @echo off>>"%HOME%yolo.bat"
-    echo python.exe %HOME%yolo-ai-cmdbot\yolo.py %%*>>"%HOME%\yolo.bat"
-    echo Created "yolo.bat" in %HOME%
-)
\ No newline at end of file
+setlocal enabledelayedexpansion
+:: Note: "~" or %HOME% is equivalent to "%HOMEDRIVE%%HOMEPATH%\" but the latter is set in VM environments (from what I can tell)
+
+:: INSTALL_DIR = Directory the "yolo-ai-cmdbot\" will go to.
+:: SCRIPT_DIR = Directory the "yolo.bat" script will go to.
+:: createDIR = Whether or not a seperate "yolo-ai-cmdbot\" directory will be made/used to hold "yolo.py" and "prompt.txt". 1=Yes, 2=Just_use_Repo (the folder this is in)
+:: createAPIKEY = Whether to create a ".openai.apikey" at %HOMEDRIVE%%HOMEPATH%\. 1=Yes, 2=No
+:: createSafetyOff = Whether to create a ".yolo-safety-off" at %HOMEDRIVE%%HOMEPATH%\. 1=Yes, 2=No
+
+:: Default values:
+set "INSTALL_DIR=%HOMEDRIVE%%HOMEPATH%\"
+set "SCRIPT_DIR=%HOMEDRIVE%%HOMEPATH%\"
+set /a "createDIR=1"
+set /a "createAPIKEY=2"
+set /a "createSafetyOff=2"
+set "installing=1"
+
+:: Set Variables To User Defined If Needed 
+:: (This was as painful to make as it looks)
+choice /n /c YNCO /m "Do you want to install with default settings (or (C)ancel) (or create (O)ptional files)? <Y/N/C/O>:"
+set installing=%ERRORLEVEL%
+goto :choice_default_!ERRORLEVEL!
+:choice_default_2
+choice /n /c YN /m "Do you want to install a seperate 'yolo-ai-cmdbot\' directory (otherwise will use this repository folder)? <Y/N>:"
+set /a createDIR=%ERRORLEVEL%
+goto :choice_use_directory_!ERRORLEVEL!
+:choice_use_directory_1
+choice /n /c NY /m "Currently 'yolo-ai-cmdbot\' will be made in '!INSTALL_DIR!' . Is this OK? <Y/N>:"
+goto :choice_install_!ERRORLEVEL!
+:choice_install_1
+set /p INSTALL_DIR="Enter a path for 'yolo-ai-cmdbot\' to be made:"
+goto :choice_use_directory_1
+:choice_install_2
+:choice_use_directory_2
+choice /n /c NY /m "Currently 'yolo.bat' will be made in '!SCRIPT_DIR!' . Is this OK? <Y/N>:"
+goto :choice_script_!ERRORLEVEL!
+:choice_script_1
+set /p SCRIPT_DIR="Enter a path for 'yolo.bat' to be made:"
+goto :choice_use_directory_2
+:choice_script_2
+:choice_default_1
+set "TARGET_DIR=!INSTALL_DIR!\yolo-ai-cmdbot\"
+set "TARGET_FULLPATH=!TARGET_DIR!\yolo.py"
+
+::Actually Install
+cls
+if /i %createDIR%==1 ( call :install_yolo_directory ) else ( call :install_yolo_repository )
+
+::Optional files:
+:choice_default_4
```

Comment:
Is it possible to skip this step if the user does a default install? 

That way the default is a quick install, and the user won't be faced with the decision about the safety mode. A first time user might not know what the safety mode means and accidentally create the file.